src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/AS_RankSumTest.java

```diff
@@ -0,0 +1,121 @@
+
+    public List<VCFInfoHeaderLine> getDescriptions() {
+        //TODO only raw for now
+//        if (AnnotationUtils.walkerRequiresRawData(callingWalker))
+            return Arrays.asList(GATKVCFHeaderLines.getInfoLine(getRawKeyName()));
+//        else
+//            return Arrays.asList(GATKVCFHeaderLines.getInfoLine(getKeyNames().get(0)));
+    }
+
+    @Override
+    public Map<String, Object> annotate(final ReferenceContext ref,
+                                         final VariantContext vc,
+                                         final Map<String, PerReadAlleleLikelihoodMap> stratifiedPerReadAlleleLikelihoodMap) {
+        return annotateRawData(ref, vc, stratifiedPerReadAlleleLikelihoodMap);
+    }
+
+    @Override
+    public Map<String, Object> annotateRawData(final ReferenceContext ref,
+                                               final VariantContext vc,
+                                               final Map<String, PerReadAlleleLikelihoodMap> perReadAlleleLikelihoodMap ) {
+        if ( perReadAlleleLikelihoodMap == null) {
+            return null;
+        }
+
+        final Map<String, Object> annotations = new HashMap<>();
+        final AlleleSpecificAnnotationData<CompressedDataList<Integer>> myData = initializeNewAnnotationData(vc.getAlleles());
+        calculateRawData(vc, perReadAlleleLikelihoodMap, myData);
+        final String annotationString = makeRawAnnotationString(vc.getAlleles(), myData.getAttributeMap());
+        if (annotationString == null){
+            return null;
+        }
+        annotations.put(getRawKeyName(), annotationString);
+        return annotations;
+    }
+
+    private AlleleSpecificAnnotationData<CompressedDataList<Integer>> initializeNewAnnotationData(final List<Allele> vcAlleles) {
+        final Map<Allele, CompressedDataList<Integer>> perAlleleValues = new HashMap<>();
+        for (final Allele a : vcAlleles) {
+            perAlleleValues.put(a, new CompressedDataList<>());
+        }
+        final AlleleSpecificAnnotationData<CompressedDataList<Integer>> ret = new AlleleSpecificAnnotationData<>(vcAlleles, perAlleleValues.toString());
+        ret.setAttributeMap(perAlleleValues);
+        return ret;
+    }
+
+    private String makeRawAnnotationString(final List<Allele> vcAlleles, final Map<Allele, CompressedDataList<Integer>> perAlleleValues) {
+        if (perAlleleValues.values().stream().allMatch(d -> d.isEmpty())){
+            return null;
+        }
+        final StringBuilder annotationString = new StringBuilder();
+        for (int i =0; i< vcAlleles.size(); i++) {
+            if (i!=0) {
+                annotationString.append(printDelim);
+            }
+            final CompressedDataList<Integer> alleleValues = perAlleleValues.get(vcAlleles.get(i));
+            annotationString.append(alleleValues);
+        }
+        return annotationString.toString();
+    }
+
+    @SuppressWarnings({"unchecked", "rawtypes"})//FIXME generics here blow up
```

Comment:
Gross! Is there no way we can properly parameterize `ReducibleAnnotationData` in the method signature?